### PR TITLE
Update Softserial documentation

### PR DESCRIPTION
**Pull request by ctzsnooze**
The old `Single Wire Software Serial` page was written a long time ago.

I've updated it and re-named it to SoftSerial.md.

The focus has been on providing an overview of softserial as it now exists, with clear recommendations to avoid using it for the radio link and for bluetooth MSP Configurator devices.

I also explain its use for GPS and GPS rescue applications, where the default GPS auto-configure option will almost certainly not work.

I also added a note in the 4.5 release notes about the change in the softserial pin assignment command, which had been overlooked.

Thanks.

### CODE REVIEW

**Review by SteveCEvans — APPROVED** [review]
The recommendations are enforced by https://github.com/betaflight/betaflight/pull/13303.

File: docs/wiki/guides/current/SoftSerial.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
+# SoftSerial
+
+Softserial provides a means to add another 'virtual' UART on MPU's with a limited number of pinned-out true UART ports.
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:3]:
```suggestion
Softserial provides a means to add another 'virtual' UART on MCU's with a limited number of pinned-out true UART ports.
```

Also would think off we only need SOFTSERIAL on F411 and perhaps F405.

File: docs/wiki/guides/current/SoftSerial.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+When the GPS is intended for GPS Rescue purposes, Betaflight requires bi-directional communication with the GPS module during the initialisation phase, so that we can configure the data rate, request only the data we need, and block any other junk that we don't need. To configure the GPS requires both Rx and Tx lines. Auto-configuration requires moving through all the available baud rates that the GPS may connect on. For most modern GPS modules, the defdault baud rate is 57600, which is too high for a two-wire Softserial setup to work with. As a result the GPS is likely to not get configured properly, and auto-configuration will fail. Even with a manually-configured GPS, with a single Rx pin on the FC for softserial set to 19200 baud and 2Hz, and with GPS auto-configuration disabled, the best you could hope for would be a very jerky GPS Rescue that may fail at any time.. Hence we strongly discourage the use of SoftSerial for GPS Rescue.
+
+### Softserial with HD OSD (displayport) over MSP
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:58]: MSP and SOFTSERIAL are no longer allowed together (in 4.5)

File: docs/wiki/guides/current/SoftSerial.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+First go to the CLI and type `resource`, to get a list of the currently assigned pin numbers for each of the currently available pads on the device.
+
+If, for instance, a `LED_STRIP 1` pin is defined to `A15`, and there is a LedStrip pad on the board, and you are not using a led strip, you could de-assign that pin from `LED_STRIP 1` with the CLI command:
+
+`resource led_strip 1 none`
+
+Then the pin can be assigned to the Softserial Port 1 Tx pin with:
+
+`resource serialTx 11 A15`
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:74]:
```suggestion
`resource SOFTSERIAL_TX 1 A15`
```

Serial TX 11 is used for LPUART

File: docs/wiki/guides/current/SoftSerial.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+If the resource reassignment is successful, a `diff` command in CLI will show:
+
+```
+resource LED_STRIP 1 NONE
+resource SERIAL_TX 11 A15
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:80]: See above

File: docs/wiki/guides/current/SoftSerial.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+For more information, see the 'Serial' page in the Development section.
+
+### Historical notes:
+
+The following notes are quite old, and may not be relevant to current versions of Betaflight
+
+CLI commands to assign softserial ports 1 and 2:
+
+`resource serial_tx 11 <pin>`
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:110]: See above

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: docs/wiki/guides/current/SoftSerial.md (hunk 1 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:3]: Thanks for noting MPU.... fixed MPU to MCU or FC as appropriate.
Mostly softserial is needed on F411, but it really depends on what UARTS are pinned out, as much as the capability of the MCU.  Added a note, `eg F411's.`

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:74]: As far as I am aware, LPUARTs are handled like normal UARTS, ie they will be configured using `serialTx` like any other port,, eg `set serialTx 6  <pin>` like any other UART, in CLI.

Serial TX 11 is not used for LPUART.  It is used to configure a softserial port.  You can confirm this easily in CLI by setting a pin to softserial as described above, then running the serial command, which will show a serial port at number 30.

File: docs/wiki/guides/current/SoftSerial.md (hunk 4 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:80]: see above

File: docs/wiki/guides/current/SoftSerial.md (hunk 5 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:110]: see above

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (haslinghuis) [reply docs/wiki/guides/current/SoftSerial.md:74]: Currently in master `UART 11` is indeed used for `LPUART`.
Softserial has it's own structure using `SOFTSERIAL_[T|R]X`

File: docs/wiki/guides/current/SoftSerial.md (hunk 5 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:121]: OK sorry it now says more clearly what the command to use in 4.5 is vs 4.4 and earlier.

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md]: That's something to be discussed elsewhere.  I twas my understanding that LPUARTs are configured using the same syntax as any other UART.

File: docs/wiki/guides/current/SoftSerial.md (hunk 4 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md]: You were right; updated the notes to show the new 4.5 form of the command, as wells the old 4.4 and earlier form.

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (haslinghuis) [reply docs/wiki/guides/current/SoftSerial.md]: That's why I had raised https://github.com/betaflight/betaflight/pull/13278 for more clarity

File: docs/wiki/guides/current/SoftSerial.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+
+:::note
+
+- When assigning softserial pins, the OK means it is compatible, and does not warrant it will work when it is configured as a part of a complete system.
+
+:::
+
+### Software Serial List of compatible pins
+
+OMNIBUS(F3) (by @jflyper)
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:141]: F3 is no longer relevant to Betaflight. Suggest this info should go to Archive.

Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:145]: You reckon it's best to move the whole original document to archive and start afresh?

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md]: Probably we just leave LPUART as TX/RX11 for 4.5.  I will add some notes.

File: docs/development/Serial.md [hunk 7]
```diff
@@ -2,19 +2,19 @@
...
+Betaflight has the concept of a function (MSP, GPS, Serial RX, etc) and a port (VCP, UARTx, SoftSerial x, LPUART1).
+Not all functions can be used on all ports due to hardware pin mapping, conflicting features, hardware, and software constraints.
 
 ## Serial Port Types
 
 - USB Virtual Com Port (VCP) - USB pins on a USB port connected directly to the processor without requiring
   a dedicated USB to UART adapter. VCP does not 'use' a physical UART port.
 - UART - A pair of dedicated hardware transmit and receive pins with signal detection and generation done in hardware.
 - SoftSerial - A pair of hardware transmit and receive pins with signal detection and generation done in software.
+- LPUART - A "Low Power" UART format available on G4 and other MCU's. By default LPUARTs are limited to 9600 Baud, but Betaflight reconfigures them to work just like a normal UART. Typically there is only one LPUART, LPUART1. It's pin assignment is normally shown as `SERIAL_TX 11` and `SERIAL_RX 11` with the `RESURCES` command.
```
Comment (nerdCopter) [inline docs/development/Serial.md:14]:
```suggestion
- LPUART - A "Low Power" UART format available on G4 and other MCU's. By default LPUARTs are limited to 9600 Baud, but Betaflight reconfigures them to work just like a normal UART. Typically there is only one LPUART, LPUART1. It's pin assignment is normally shown as `SERIAL_TX 11` and `SERIAL_RX 11` with the `RESOURCE` command.
```

File: docs/wiki/guides/current/SoftSerial.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+- `resource show all` in CLI will provide a full list of all serial ports, timers etc.
+- just because the resources are re-assigned, it does not mean that the Softserial port will work properly.
+
+Oscarliang provides a good summary of [how to set up softserial](https://oscarliang.com/betaflight-soft-serial/). Remember that the commands have changed in Betaflight 4.5.
+
+### The CLI Serial command.
+
+When Softserial is active, the `serial` CLI command will show all configured serial ports. A line starting with 30 refers to Softserial port 1, and one starting with 31 refers to Softserial port 2.
+
+For example, if Softserial is configured for smartAudio on Softserial port 1 Tx pin, a line like this should appear when the `serial` command is sent:
```
Comment (nerdCopter) [inline docs/wiki/guides/current/SoftSerial.md:114]:
```suggestion
For example, if Softserial is configured for SmartAudio on Softserial port 1 Tx pin, a line like this should appear when the `serial` command is sent:
```

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (haslinghuis) [reply docs/wiki/guides/current/SoftSerial.md]: - CLI in firmware does not use the same definition which we use to add LPUART1 in config.
- If manufacturers would add  UART11 then we have a problem.
- LPUART has range 40-49 and should not use UART 1-20 range.
- UART 11 in firmware does not exist. It's called LPUART1

```
configs/AIRBOTG4AIO/config.h:63:#define LPUART1_TX_PIN       PB11
configs/AIRBOTG4AIO/config.h:69:#define LPUART1_RX_PIN       PB10
configs/KAKUTEG4AIO/config.h:54:#define LPUART1_TX_PIN       PB11
configs/KAKUTEG4AIO/config.h:59:#define LPUART1_RX_PIN       PB10
configs/MAMBAG4/config.h:71:#define LPUART1_RX_PIN       PB11
configs/TAKERG4AIO/config.h:50:#define LPUART1_TX_PIN      PB10
configs/TAKERG4AIO/config.h:51:#define LPUART1_RX_PIN      PB11
configs/LUXHDAIO-G4/config.h:51:#define LPUART1_TX_PIN       PB11
configs/LUXHDAIO-G4/config.h:56:#define LPUART1_RX_PIN       PB10
configs/LUXMICROAIO/config.h:56:#define LPUART1_TX_PIN       PB11
configs/LUXMICROAIO/config.h:62:#define LPUART1_RX_PIN       PB10
```

File: docs/development/Serial.md (hunk 7 above)
Comment (ctzsnooze) [reply docs/development/Serial.md:14]: Thanks very much for proof-reading so carefully

File: docs/wiki/guides/current/SoftSerial.md (hunk 8 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md:114]: thanks, sorry I missed that

File: docs/wiki/guides/current/SoftSerial.md (hunk 6 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md]: done!

File: docs/wiki/guides/current/SoftSerial.md (hunk 3 above)
Comment (ctzsnooze) [reply docs/wiki/guides/current/SoftSerial.md]: Made notes about TX11 being for LPUART1 and explained some stuff about LPUART generally in the release notes also

**Comment by ctzsnooze** [comment]
@haslinghuis I would be grateful if this could be approved and merged.  I do appreciate that there is a PR of yours about serial.md.  This PR adds a bit more detail about LPUART.  When we finalise how LPUART is handled - noting your recent PR for Resource command and LPUART - then we can make some updates.  But we need a useful softserial document now, and this is a big improvement on what we had.

File: docs/development/Serial.md [hunk 9]
```diff
@@ -2,19 +2,19 @@
...
 - UART - A pair of dedicated hardware transmit and receive pins with signal detection and generation done in hardware.
 - SoftSerial - A pair of hardware transmit and receive pins with signal detection and generation done in software.
+- LPUART - A "Low Power" UART format available on G4 and other MCU's. By default LPUARTs are limited to 9600 Baud, but Betaflight reconfigures them to work just like a normal UART. Typically there is only one LPUART, LPUART1. It's pin assignment is normally shown as `SERIAL_TX 11` and `SERIAL_RX 11` with the `RESOURCE` command.
 
-UART is the most efficient in terms of CPU usage.
-SoftSerial is the least efficient and slowest, SoftSerial should only be used for low-bandwidth usages, such as telemetry transmission.
+A "real" UART is the most efficient in terms of CPU usage.
+SoftSerial is the least efficient and slowest. SoftSerial should only be used for low-bandwidth, low-priority applications, such as sending or receiving telemetry data.
 
 UART ports are sometimes exposed via on-board USB to UART converters, such as the CP2102 as found on the Naze and Flip32 boards.
```
Comment (SteveCEvans) [inline docs/development/Serial.md:19]: This paragraph refers to ancient history and is no longer needed.

**Review by SteveCEvans — CHANGES_REQUESTED** [review]

File: docs/development/Serial.md (hunk 9 above)
Comment (ctzsnooze) [reply docs/development/Serial.md]: Removed, and the text clarified.

File: docs/wiki/guides/current/SoftSerial.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+If auto-configuration is disabled, the GPS must be manually configured to boot up with the following settings:
+
+- baud rate 19200
+- send Nav_PVT data, nothing else, at 2 Hz.
+  It can then be connected to a single Softserial Rx pin on the FC at 19200 baud. With only 2Hz data rates, rescues may be very jerky and unreliable. Don't take chances, use a real UART.
+
+### Softserial with HD OSD (displayport) over MSP
+
+In 4.5, the user of softserial for MSP connections, including for HD OSD's is not permitted.
```
Comment (haslinghuis) [inline docs/wiki/guides/current/SoftSerial.md:76]:
```suggestion
In 4.5, the use of softserial for MSP connections, including for HD OSD's is not permitted.
```

**Review by nerdCopter — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**